registry/remote/repository_test.go

```diff
@@ -7447,3 +7447,24 @@ func TestRepository_do(t *testing.T) {
 		t.Errorf("Repository.do() = %v, want %v", gotWarnings, wantWarnings)
 	}
 }
+
+func TestRepository_clone(t *testing.T) {
+	repo, err := NewRepository("localhost:1234/repo/image")
+	if err != nil {
+		t.Fatalf("invalid repository: %v", err)
+	}
+
+	crepo := repo.clone()
+
+	if repo.Reference != crepo.Reference {
+		t.Fatal("references should be the same")
+	}
+
+	if !reflect.DeepEqual(&repo.referrersPingLock, &crepo.referrersPingLock) {
+		t.Fatal("referrersPingLock should be copied")
```

Comment:
This looks a bit misleading. Should the mutex not be copied?

Reply:
I clarified the message.